pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -40,6 +49,94 @@ func (h *AnnotationBackend) RecordStatesAsync(ctx context.Context, rule *ngmodel
 	go h.recordAnnotationsSync(ctx, panel, annotations, logger)
 }
 
+func (h *AnnotationBackend) QueryStates(ctx context.Context, query models.HistoryQuery) (*data.Frame, error) {
```

Comment:
The rest of this file maps the query interface onto annotations, as best it can. We error or warn if a query attempts to use a query feature that annotations do not fundamentally support.

This logic isn't used by anything currently, and is meant to be more of a reference implementation. We expect it will change over time as the query interface is implemented for the other backend types.